src/Config/ConfigPanel.ts

```diff
@@ -0,0 +1,98 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+import {getNonce} from './GetNonce';
+
+
+export class ConfigPanel {
+  /**
+   * Track the current panel. Only allow a single panel to exist at a time.
+   */
+  public static currentPanel: ConfigPanel|undefined;
+
+  public static readonly viewType = 'one-vscode';
+
+  private readonly _panel: vscode.WebviewPanel;
+  private readonly _extensionUri: vscode.Uri;
+  private _disposables: vscode.Disposable[] = [];
+
+  public static createOrShow(context: vscode.ExtensionContext) {
+    const column =
+        vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined;
+
+    // If we already have a panel, show it.
+    if (ConfigPanel.currentPanel) {
+      ConfigPanel.currentPanel._panel.reveal(column);
+      ConfigPanel.currentPanel._update(context);
+      return;
+    }
+
+    // Otherwise, create a new panel.
+    const panel = vscode.window.createWebviewPanel(
+        ConfigPanel.viewType, 'ConfigurationSettings', column || vscode.ViewColumn.One, {
+          // Enable javascript in the webview
+          enableScripts: true,
+          localResourceRoots: [],
+        });
+
+    ConfigPanel.currentPanel = new ConfigPanel(panel, context);
+  }
+
+  public static kill() {
+    ConfigPanel.currentPanel ?.dispose();
+    ConfigPanel.currentPanel = undefined;
+  }
+
+  public static revive(panel: vscode.WebviewPanel, context: vscode.ExtensionContext) {
+    ConfigPanel.currentPanel = new ConfigPanel(panel, context);
+  }
+
+  private constructor(panel: vscode.WebviewPanel, context: vscode.ExtensionContext) {
+    this._panel = panel;
+    this._extensionUri = context.extensionUri;
+
+    // Set the webview's initial html content
+    this._update(context);
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+  }
+
+  public dispose() {
+    ConfigPanel.currentPanel = undefined;
+
+    // Clean up our resources
+    this._panel.dispose();
+
+    while (this._disposables.length) {
+      const x = this._disposables.pop();
+      if (x) {
+        x.dispose();
+      }
+    }
+  }
+
+  private async _update(context: vscode.ExtensionContext) {
+    const webview = this._panel.webview;
+    webview.html = this._getHtmlForWebview(webview, context);
+  }
+
+  private _getHtmlForWebview(webview: vscode.Webview, context: vscode.ExtensionContext) {
+    // Use a nonce to only allow specific scripts to be run
+    const nonce = getNonce();
+
+    return '';
```

Comment:
please remove `nonce` as this is not used as of now.
please add `TODO ` with some comments explaining your plan.

Reply:
I will remove nonce.
and i will add `TODO` in our draft #117

Reply:
> and i will add TODO in our draft #117

Why add to draft?

Reply:
Because i think plan means how pull request will be done.
So draft is good place for explaining our plan for pull request.
If you have a better idea, please share me.
And if your intention is different from i guess, then could you explain it more specific please.

Reply:
If you are going to land `return '';` code as-is
- as we discussed before this is not the real code and need to be changed to real working code
- in that sense, it would be better to add `TODO` comment here
- so that when others see this code in the master branch, they will know that this code will be changed someday

I've added a note at https://github.com/Samsung/ONE-vscode/pull/122#discussion_r714520864 but I think you didn't understand what I meant.

Reply:
If you are going to change `return '';` in this PR to working code
- to make communication clear, please add `PR/NO MERGE` tag
- update the code
- remove  `PR/NO MERGE` tag and ping reviewers again

Reply:
Ah, i didn't understand what you mean.
Then, i think getting rid of `_getHtmlForWebview` part is better option than add `TODO` on this page.